stdlib/public/runtime/Casting.cpp

```diff
@@ -372,9 +372,7 @@ swift::swift_getTypeName(const Metadata *type, bool qualified) {
     auto name = nameForMetadata(type, qualified);
     // Copy it to memory we can reference forever.
     auto size = name.size();
-    auto result = (char *)malloc(size + 1);
-    memcpy(result, name.data(), size);
-    result[size] = 0;
+    auto result = strndup(name.data(), size);
```

Comment:
Also it's less portable (though maybe this section of the file is already Apple-only, didn't check).